core/schema/query.go

```diff
@@ -49,3 +57,61 @@ func (s *querySchema) pipeline(ctx *router.Context, parent *core.Query, args pip
 	})
 	return parent, nil
 }
+
+type checkVersionCompatibilityArgs struct {
+	Version string
+}
+
+func (s *querySchema) checkVersionCompatibility(ctx *router.Context, _ *core.Query, args checkVersionCompatibilityArgs) (bool, error) {
+	recorder := progrock.RecorderFromContext(ctx)
+
+	// Skip development version
+	if strings.Contains(engine.Version, "devel") {
+		recorder.Warn("Using Engine in development version, skip version compatibility check.")
+
+		return true, nil
+	}
+
+	engineVersion, err := semver.Parse(engine.Version)
+	if err != nil {
+		recorder.Error("Could not compare engine and SDK version, they might be incompatible!")
+
+		// TODO: throw an error and abort the session
+		// return false, err
+		return true, nil
+	}
+
+	sdkVersion, err := semver.Parse(args.Version)
+	if err != nil {
+		recorder.Error("Could not compare engine and SDK version, they might be incompatible!")
+
+		// TODO: throw an error and abort the session
+		// return false, err
+		return true, nil
+	}
```

Comment:
Maybe I'm missing something, but I would expect both of these values to be valid semver (except for `devel` as handled above), so I think it's better to just `return false, err` if they're not.

Reply:
you're right!